tests/unittests/threed/object_3d_tests.py

```diff
@@ -314,3 +314,491 @@ def test_traverse(self):
         self.assertListEqual(
             list(sub.traverse(include_self=False)), objects[2:4]
         )
+
+
+class TestObject3DRemove(unittest.TestCase):
```

Comment:
lot of tests, thanks! Looks like AI written tests lol, but if you did it all manually, good on you sir